packages/python/src/components/TypeExpression.tsx

```diff
@@ -0,0 +1,71 @@
+import { Children, For, memo } from "@alloy-js/core";
+
+export interface SingleTypeExpressionProps {
+  children: Children;
+}
+
+export function SingleTypeExpression(props: SingleTypeExpressionProps) {
+  const resolvedChildren = memo(() => props.children);
+
+  return (
+    <group>
+      <indent>
+        <sbr />
+        {resolvedChildren}
+      </indent>
+      <sbr />
+    </group>
+  );
+}
+
+export interface UnionTypeExpressionProps {
+  children: SingleTypeExpressionProps[];
+  optional?: boolean;
+}
+
+export function UnionTypeExpression(props: UnionTypeExpressionProps) {
+  // Map each SingleTypeExpressionProps to a SingleTypeExpression element (Children)
+  let childrenElements: Children[] = props.children.map((childProps) => (
+    <SingleTypeExpression {...childProps} />
+  ));
+
+  if (props.optional) {
+    childrenElements = [...childrenElements, "None"];
+  }
+
+  return (
+    <group>
+      <ifBreak>(</ifBreak>
+      <indent>
+        <sbr />
+        <For
+          each={childrenElements}
+          joiner={
+            <>
+              <br />|{" "}
+            </>
+          }
+        >
+          {(child) => child}
+        </For>
+      </indent>
+      <sbr />
+      <ifBreak>)</ifBreak>
+    </group>
+  );
+}
+
+export type TypeExpressionProps =
```

Comment:
I don't see `TypeProps` defined in this file; was it renamed to this?

Reply:
Yeah, precisely. And seems like I left some references behind and no tests failed.